packages/bpk-component-input/src/BpkInput.js

```diff
@@ -72,17 +92,48 @@ const BpkInput = props => {
     classNames.push(getClassName('bpk-input--docked-last'));
   }
   if (className) {
-    classNames.push(className);
+    if (clearable) {
+      containerClassNames.push(className);
+    } else {
+      classNames.push(className);
+    }
   }
 
-  return (
+  const renderedInput = (
     <input
       className={classNames.join(' ')}
-      ref={inputRef}
+      ref={input => {
+        ref = input;
+        if (inputRef) {
+          inputRef(input);
+        }
+      }}
       aria-invalid={isInvalid}
+      value={value}
       {...rest}
     />
   );
+
+  return clearable ? (
+    <div className={containerClassNames.join(' ')}>
+      {renderedInput}
+      {value.length > 0 && (
```

Comment:
I like the logic which only shows the button when there is text and it's selected, but it seems a bit odd that the button still works even when it's hidden

Reply:
Maybe we could capture all this logic in state, and then only render a button at all when it is focused.


Then again, the issue may be it not showing when not focused.

Even when not focused, I would expect the button to be available.
![img_f343cb69a02c-1](https://user-images.githubusercontent.com/30267516/35433674-34966ac6-027c-11e8-9c83-2428b242181d.jpeg)

Reply:
@matteo-hertel thanks for all the feedback - have addressed it all!